bin/utest/pipe.c

```diff
@@ -104,3 +107,233 @@ int test_pipe_child_signaled(void) {
 
   return 0;
 }
+
+int test_pipe_blocking_flag_manipulation(void) {
+  int pipe_fd[2];
+
+  /* creating pipe */
+  int pipe2_ret = pipe2(pipe_fd, O_NONBLOCK);
+  assert(pipe2_ret == 0);
+
+  /* check if flag is set */
+  int is_flag_set;
+  is_flag_set = fcntl(pipe_fd[0], F_GETFL) & O_NONBLOCK;
+  assert(is_flag_set);
+  is_flag_set = fcntl(pipe_fd[1], F_GETFL) & O_NONBLOCK;
+  assert(is_flag_set);
+
+  /* unset same flag for read end */
+  int read_flagset_with_block = fcntl(pipe_fd[0], F_GETFL);
+  assert(read_flagset_with_block > 0);
+  fcntl(pipe_fd[0], F_SETFL, read_flagset_with_block & ~O_NONBLOCK);
+  /* unset same flag for write end */
+  int write_flagset_with_block = fcntl(pipe_fd[1], F_GETFL);
+  assert(write_flagset_with_block > 0);
+  fcntl(pipe_fd[1], F_SETFL, write_flagset_with_block & ~O_NONBLOCK);
+
+  /* check if flag is not set */
+
+  int is_flag_not_set;
+  is_flag_not_set = fcntl(pipe_fd[0], F_GETFL) & O_NONBLOCK;
+  assert(!is_flag_not_set);
+  is_flag_not_set = fcntl(pipe_fd[1], F_GETFL) & O_NONBLOCK;
+  assert(!is_flag_not_set);
+
+  close(pipe_fd[0]);
+  close(pipe_fd[1]);
+
+  return 0;
+}
+
+int test_pipe_write_interruptible_sleep(void) {
+  int pipe_fd[2];
+  pid_t child_pid;
+
+  /* creating pipe */
+  int pipe2_ret = pipe2(pipe_fd, 0);
+  assert(pipe2_ret == 0);
+
+  /* forking */
+  child_pid = fork();
+  assert(child_pid >= 0);
+
+  if (child_pid == 0) { /* child */
+    close(pipe_fd[0]);  /* closing read end of pipe */
+
+    struct sigaction sa = {
+      .sa_handler = sigpipe_handler,
+      .sa_flags = 0,
+    };
+
+    sigaction(SIGALRM, &sa, NULL);
+
+    int page_size = 4096; // based on implementation of getpagesize()
+    char data[page_size];
```

Comment:
You have inconsistency in your code. Here you assume that page size is 4096 (it's true for us), but latter you call `getpagesize`. Please do that in one way.

Also I'm not a fan of VAL and I suggest to avoid them.

Reply:
I also don't like VAL, but return value of `getpagesize()` may change at some point and that test should be ready for that.